File: common/src/main/java/taboolib/common/env/runtime/RuntimeDependency.java
package taboolib.common.env.runtime;

import java.lang.annotation.*;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(RuntimeDependencies.class)
public @interface RuntimeDependency {

    String group();

    String id();

    String version();

    String hash();

    String repository() default "https://maven.aliyun.com/repository/central";

}

File: module-dependency/src/main/kotlin/taboolib/module/dependency/RuntimeDependency.kt
package taboolib.module.dependency

@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@kotlin.annotation.Repeatable
annotation class RuntimeDependency(
    val group: String,
    val id: String,
    val version: String,
    val hash: String,
    val repository: String = "https://maven.aliyun.com/repository/central"
)